### PR TITLE
Add Person Reidentification Feature

### PR DESCRIPTION
**Pull request by HoukChen**
@LewisLiuPub Person reidentification feature is enabled, please review.

### CODE REVIEW

**Comment by houk-intel** [comment]
@LewisLiuPub  Parameter confidence_threshold is activated for the person_reid inference now.

File: dynamic_vino_lib/src/inferences/person_reidentification.cpp [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+bool dynamic_vino_lib::PersonReidentification::fetchResults()
+{
+  bool can_fetch = dynamic_vino_lib::BaseInference::fetchResults();
+  if (!can_fetch) {return false;}
+  bool found_result = false;
+  InferenceEngine::InferRequest::Ptr request = getEngine()->getRequest();
+  std::string output = valid_model_->getOutputName();
+  const float * output_values = request->GetBlob(output)->buffer().as<float *>();
+  for (int i = 0; i < getResultsLength(); i++) {
```
[the remark this answers is not in the document]
Comment (houk-intel) [reply dynamic_vino_lib/src/inferences/person_reidentification.cpp]: It's updated in the findMatchPerson function

File: dynamic_vino_lib/src/pipeline_manager.cpp [hunk 2]
```diff
@@ -285,6 +289,23 @@ PipelineManager::createObjectSegmentation(const Params::ParamManager::InferenceP
...
+PipelineManager::createPersonReidentification(
+  const Params::ParamManager::InferenceParams & infer)
+{
+  auto person_reidentification_model =
+    std::make_shared<Models::PersonReidentificationModel>(infer.model, 1, 1, infer.batch);
+  person_reidentification_model->modelInit();
+  auto person_reidentification_engine = std::make_shared<Engines::Engine>(
+    plugins_for_devices_[infer.engine], person_reidentification_model);
+  auto reidentification_inference_ptr =
+    std::make_shared<dynamic_vino_lib::PersonReidentification>(0.7);
```
[the remark this answers is not in the document]
Comment (houk-intel) [reply dynamic_vino_lib/src/pipeline_manager.cpp]: It's now configurable in the yaml file

File: dynamic_vino_lib/src/inferences/person_reidentification.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (houk-intel) [reply dynamic_vino_lib/src/inferences/person_reidentification.cpp]: It is implemented in this way in the sample code. Should we move it outside this function?

[the remark this answers is not in the document]
Comment (LewisLiuPub) [reply dynamic_vino_lib/src/inferences/person_reidentification.cpp]: needn't, we keep the implementation just now.

**Pull request merged**